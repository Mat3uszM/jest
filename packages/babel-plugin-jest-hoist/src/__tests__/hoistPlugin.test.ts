import pluginTester from 'babel-plugin-tester';
import babelPluginJestHoist from '..';

pluginTester({
  plugin: babelPluginJestHoist,
  pluginName: 'babel-plugin-jest-hoist',
  tests: {
    'top level mocking': {
      code: `
        require('x');

        jest.enableAutomock();
        jest.disableAutomock();
      `,
      snapshot: true,
    },
    'within a block': {
      code: `
        beforeEach(() => {
          require('x')
          jest.mock('someNode')
        })
      `,
      snapshot: true,
    },
    'within a block with no siblings': {
      code: `
        beforeEach(() => {
          jest.mock('someNode')
        })
      `,
      snapshot: true,
    },
  },
});
