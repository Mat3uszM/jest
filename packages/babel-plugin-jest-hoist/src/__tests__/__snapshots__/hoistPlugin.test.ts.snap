// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-jest-hoist top level mocking: top level mocking 1`] = `

require('x');

jest.enableAutomock();
jest.disableAutomock();

      ↓ ↓ ↓ ↓ ↓ ↓

_getJestObj().enableAutomock();

_getJestObj().disableAutomock();

function _getJestObj() {
  const { jest } = require("@jest/globals");

  _getJestObj = () => jest;

  return jest;
}

require("x");


`;

exports[`babel-plugin-jest-hoist within a block with no siblings: within a block with no siblings 1`] = `

beforeEach(() => {
  jest.mock('someNode')
})

      ↓ ↓ ↓ ↓ ↓ ↓

function _getJestObj() {
  const { jest } = require("@jest/globals");

  _getJestObj = () => jest;

  return jest;
}

beforeEach(() => {
  _getJestObj().mock("someNode");
});


`;

exports[`babel-plugin-jest-hoist within a block: within a block 1`] = `

beforeEach(() => {
  require('x')
  jest.mock('someNode')
})

      ↓ ↓ ↓ ↓ ↓ ↓

function _getJestObj() {
  const { jest } = require("@jest/globals");

  _getJestObj = () => jest;

  return jest;
}

beforeEach(() => {
  _getJestObj().mock("someNode");

  require("x");
});


`;
